src/deep-cache/lib/Driver/CloudFrontDriver.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Created by CCristi <ccovali@mitocgroup.com> on 12/10/15.
+ */
+
+'use strict';
+
+import {AbstractFsDriver} from './AbstractFsDriver';
+import {MethodNotAvailableException} from '../Exception/MethodNotAvailableException';
+import {Exception} from '../Exception/Exception';
+
+/**
+ * CloudFront Cache Driver
+ */
+export class CloudFrontDriver extends AbstractFsDriver {
+
+  /**
+   * @todo: Find better way to extract the working microservice identifier
+   *
+   * @param {ContainerAware} containerAware
+   * @param {String} directory
+   */
+  constructor(containerAware, directory = AbstractFsDriver.DEFAULT_DIRECTORY) {
+    super(directory);
+
+    this._containerAware = containerAware;
+  }
+
+  /**
+   * @param {String} key
+   * @param {Function} callback
+   * @private
+   */
+  _has(key, callback = () => {}) {
+    this._get(key, (err, data) => {
+      callback(err, data !== null);
+    });
+  }
+
+  /**
+   * @param {String} key
+   * @param {Function} callback
+   * @private
+   */
+  _get(key, callback = () => {}) {
+    this._request(key, (err, data) => {
+      if (err) {
+        return callback(err, null);
+      }
+
+      try {
+        let parsedData = JSON.parse(data);
+
+        if (parsedData.expires && parsedData.expires < AbstractFsDriver._now) {
+          throw new Exception('Expired');
+        }
+
+        callback(null, parsedData.value);
+      } catch (e) {
+        // @todo: Find a way to invalidate broken or expired keys
+        callback(null, null);
+      }
+    });
+  }
+
+  /**
+   * @param {String} key
+   * @param {Object} value
+   * @param {Function} callback
+   * @private
+   */
+  _set(key, value, callback = () => {}) {
+    throw new MethodNotAvailableException('set');
+  }
+
+  /**
+   * @param key
+   * @param timeout
+   * @param callback
+   * @private
+   */
+  _invalidate(key, timeout = 0, callback = () => {}) {
+    throw new MethodNotAvailableException('invalidate');
+  }
+
+  /**
+   * @param {String} key
+   * @returns {String}
+   * @private
+   */
+  _buildKey(key) {
+    return this._asset.locate(`@${this._microservice}:${super._buildKey(key)}`);
+  }
+
+  /**
+   * @param {String} url
+   * @param {Function} callback
+   * @private
+   */
+  _request(url, callback) {
+    var client = new XMLHttpRequest();
+
+    client.onreadystatechange = function() {
+      if (client.readyState == 4) {
+        if (client.status != 200) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/MitocGroup/deep-framework/file/1585049076/issues/source?bid=2895475&fileBranchId=3024729#l104)